cibyl/sources/zuul/transactions/__init__.py

```diff
@@ -358,6 +365,87 @@ def get(self):
         return [BuildResponse(build) for build in builds]
 
 
+class TestsRequest(Request):
+    """High-Level petition focused on retrieval of data related to test
+    cases.
+    """
+
+    def __init__(self, build: ZuulBuildAPI):
+        """Constructor.
+
+        :param build: Low-Level API to the build to get the test cases from.
+        """
+        super().__init__()
+
+        self._build = build
+
+    def with_name(self, *pattern: str) -> 'TestsRequest':
+        """Will limit request to tests whose name follows a certain pattern.
+
+        :param pattern: Regex pattern for the desired test name.
+        :return: The request's instance.
+        """
+
+        def test(response):
+            return any(
+                matches_regex(patt, response.name) for patt in pattern
```

Comment:
Maybe `for pattern in patterns`?

Reply:
I saw somewhere to name * variables in singular, like in: *pattern. That decision has led me to this thing here. The whole file uses this convention, so I am not going to bother changing such thing. I will most likely not do it again next time though.

Reply:
If you're not going to do it anymore then let's comment it with the team. At this way all of we have the same pattern.

Thank you, Jorge.